### ISSUE TITLE
name 'bsz' is not defined for LLaVA-OneVision-1.5-4B

### ISSUE THREAD

**Issue by cook000** [EvolvingLMMs-Lab/LLaVA-OneVision-1.5#100]
I encountered a NameError when attempting to access attention maps in Eager mode. It appears that the variable bsz (Batch Size) is referenced in the model's forward pass or attention visualization logic but is not defined in certain code paths.  path:  huggingface/modules/transformers_modules/model_hyphen_llava_hyphen_onevision_hyphen_4b/modeling_llavaonevision1_5.py

**Comment by yiyexy** [comment]
Could you provide more details about this problem? Maybe a log?

**Comment by cook000** [comment]
下面的部分huggingface/modules/transformers_modules/model_hyphen_llava_hyphen_onevision_hyphen_4b/modeling_llavaonevision1_5.py里的，bsz没有定义。class LLaVAOneVision1_5_Attention(nn.Module):
    """
    Multi-headed attention from 'Attention Is All You Need' paper. Modified to use sliding window attention: Longformer
    and "Generating Long Sequences with Sparse Transformers".
    """

    def __init__(self, config: LLaVAOneVision1_5_TextConfig, layer_idx: Optional[int] = None):
        super().__init__()
        self.config = config
        self.layer_idx = layer_idx
        self.head_dim = getattr(config, "head_dim", config.hidden_size // config.num_attention_heads)
        self.num_key_value_groups = config.num_attention_heads // config.num_key_value_heads
        self.scaling = self.head_dim**-0.5
        self.attention_dropout = config.attention_dropout
        self.is_causal = True

        self.q_proj = nn.Linear(
            config.hidden_size, config.num_attention_heads * self.head_dim, bias=config.attention_bias
        )
        self.k_proj = nn.Linear(
            config.hidden_size, config.num_key_value_heads * self.head_dim, bias=config.attention_bias
        )
        self.v_proj = nn.Linear(
            config.hidden_size, config.num_key_value_heads * self.head_dim, bias=config.attention_bias
        )
        self.o_proj = nn.Linear(
            config.num_attention_heads * self.head_dim, config.hidden_size, bias=config.attention_bias
        )
        self.q_norm = LLaVAOneVision1_5_RMSNorm(self.head_dim, eps=config.rms_norm_eps)  # unlike olmo, only on the head dim!
        self.k_norm = LLaVAOneVision1_5_RMSNorm(self.head_dim, eps=config.rms_norm_eps)  # thus post q_norm does not need reshape
        self.sliding_window = config.sliding_window if config.layer_types[layer_idx] == "sliding_attention" else None

    def forward(
        self,
        hidden_states: torch.Tensor,
        position_embeddings: Optional[Tuple[torch.Tensor, torch.Tensor]] = None,
        attention_mask: Optional[torch.Tensor] = None,
        past_key_value: Optional[Cache] = None,
        cache_position: Optional[torch.LongTensor] = None,
        position_ids: Optional[torch.LongTensor] = None,
        output_attentions: bool = False,
        use_cache: bool = False,
        **kwargs: Unpack[FlashAttentionKwargs],
    ) -> Tuple[torch.Tensor, Optional[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
        input_shape = hidden_states.shape[:-1]
        hidden_shape = (*input_shape, -1, self.head_dim)

        query_states = self.q_norm(self.q_proj(hidden_states).view(hidden_shape)).transpose(1, 2)
        key_states = self.k_norm(self.k_proj(hidden_states).view(hidden_shape)).transpose(1, 2)
        value_states = self.v_proj(hidden_states).view(hidden_shape).transpose(1, 2)

        cos, sin = position_embeddings
        query_states, key_states = apply_rotary_pos_emb(query_states, key_states, cos, sin)

        if past_key_value is not None:
            # sin and cos are specific to RoPE models; cache_position needed for the static cache
            cache_kwargs = {"sin": sin, "cos": cos, "cache_position": cache_position}
            key_states, value_states = past_key_value.update(key_states, value_states, self.layer_idx, cache_kwargs)
        
        key_states = repeat_kv(key_states, self.num_key_value_groups)
        value_states = repeat_kv(value_states, self.num_key_value_groups)

        attn_weights = torch.matmul(query_states, key_states.transpose(2, 3)) / math.sqrt(self.head_dim)

        if attention_mask is not None:  # no matter the length, we just slice it
            causal_mask = attention_mask[:, :, :, : key_states.shape[-2]]
            attn_weights = attn_weights + causal_mask

        # Replace inf values with zeros in attention weights to prevent NaN propagation
        if query_states.dtype == torch.float16:
            attn_weights = torch.where(torch.isinf(attn_weights), torch.zeros_like(attn_weights), attn_weights)

        # upcast attention to fp32
        attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(query_states.dtype)
        attn_weights = nn.functional.dropout(attn_weights, p=self.attention_dropout, training=self.training)
        attn_output = torch.matmul(attn_weights, value_states)

        if attn_output.size() != (bsz, self.num_heads, input_shape[1], self.head_dim):
            raise ValueError(
                f"`attn_output` should be of size {(bsz, self.num_heads, input_shape[1], self.head_dim)}, but is"
                f" {attn_output.size()}"
            )

        attn_output = attn_output.transpose(1, 2).contiguous()
        attn_output = attn_output.reshape(*input_shape, -1)

        attn_output = self.o_proj(attn_output)

        if not output_attentions:
            attn_weights = None

        return attn_output, attn_weights, past_key_value

**Comment by yiyexy** [comment]
感谢您的反馈，`bsz=hidden_states.shape[0]`，请您先自行修补，后面我们将解决这个问题

**Comment by cook000** [comment]
感谢！我尝试在自己的代码里修补了这个问题，但是似乎会自动检查更新而下载。关闭更新后遇到了其他的问题。导致我始终没法画出模型的注意力图。如果可以的话，我能问问您和团队大概会在什么时候修补这个漏洞吗？




🐱
***@***.***

**Comment by yiyexy** [comment]
我已修正该问题，非常感谢您的反馈。

**Comment by cook000** [comment]
感谢您的及时处理! 这对我来说非常重要，thank you🙏🏻



---Original---
From: ***@***.***&gt;
Date: Fri, Feb 6, 2026 23:50 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [EvolvingLMMs-Lab/LLaVA-OneVision-1.5] name 'bsz' is not definedfor LLaVA-OneVision-1.5-4B (Issue #100)


yiyexy left a comment (EvolvingLMMs-Lab/LLaVA-OneVision-1.5#100)
 
我已修正该问题，非常感谢您的反馈。
